File: windows/src/GoalMaker.Core/Design/ThemeDefinition.cs
namespace GoalMaker.Core.Design;

/// <summary>One of the switchable themes (ADR 0008), with its palettes for light, dark and pure black.</summary>
public sealed record ThemeDefinition(
    string Id,
    string Name,
    string Summary,
    ThemeTypography Typography,
    ThemeShapes Shapes,
    Palette Light,
    Palette Dark,
    Palette Black);

File: android/app/src/main/kotlin/com/goalmaker/app/domain/design/ThemeDefinition.kt
package com.goalmaker.app.domain.design

/** One of the switchable themes (ADR 0008), with its palettes for light, dark and pure black. */
data class ThemeDefinition(
    val id: String,
    val name: String,
    val summary: String,
    val typography: ThemeTypography,
    val shapes: ThemeShapes,
    val light: Palette,
    val dark: Palette,
    val black: Palette,
)
